spec/heartcheck/checks/newrelic_spec.rb

```diff
@@ -0,0 +1,25 @@
+RSpec.describe Heartcheck::Checks::Newrelic do
+  describe '#validate' do
+    it 'does not record an error when everything is fine' do
+      allow(NewRelic::Agent).to receive(:notify_error)
```

Comment:
I am almost sure you are stubing a method which does not exists.

I've just learned it is possible to configure rspec to raise error in these conditions:

https://makandracards.com/makandra/36409-configure-rspec-to-raise-an-error-when-stubbing-a-non-existing-method

It would raise an error like this:
```ruby
NewRelic::Agent does not implement: notify_error
```